src/basic/conf-virt.c

```diff
@@ -0,0 +1,290 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#if defined(__i386__) || defined(__x86_64__)
+#include <cpuid.h>
+#endif
+#include <errno.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "conf-virt.h"
+#include "string-table.h"
+
+#define CPUID_PROCESSOR_INFO_AND_FEATURE_BITS 0x1
+
+/*
+ * AMD64 Architecture Programmer’s Manual Volume 3:
+ * General-Purpose and System Instructions.
+ * Chapter: E4.1 - Maximum Extended Function Number and Vendor String
+ *  https://www.amd.com/system/files/TechDocs/24594.pdf
+ */
+#define CPUID_GET_HIGHEST_FUNCTION 0x80000000
```

Comment:
hmm, this is a signed int constant that is actually not a valid signed int value, because larger than 2^31-1. Please define with a an `U` suffix. Or even better, use `UINT32_C(…)`

(similar below)